#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/tables/protos/Table.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/factory/fire_extinguisher/protos/FireExtinguisher.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/factory/containers/protos/WoodenBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/apartment_structure/protos/Door.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/apartment_structure/protos/Wall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/school_furniture/protos/Blackboard.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/cabinet/protos/Cabinet.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/factory/tools/protos/PaintBucket.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/VarnishedPine.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/tables/protos/Desk.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/solids/protos/SolidPipe.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/freight/protos/IntermodalOfficeContainer.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/humans/pedestrian/protos/Pedestrian.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/computers/protos/Monitor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/chairs/protos/OfficeChair.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/universal_robots/protos/UR5e.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/devices/robotiq/protos/Robotiq3fGripper.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
      material1 "arena"
    }
  ]
}
Viewpoint {
  orientation 0.3159136141111937 -0.24667682847431413 -0.9161599918753589 4.589010505442715
  position 2.605864878855678 -2.3488108302111748 5.231630894189279
}
TexturedBackground {
  texture "factory"
}
TexturedBackgroundLight {
  texture "factory"
}
Pedestrian {
  translation 6.01814 0.289776 1.27
  rotation 0 0 1 1.0472
}
Desk {
  translation 2.74452 3.11582 -0.00919876491348992
  rotation 0.7473689530319714 -0.6487840834508276 -0.14324406132459577 4.4347260932278976e-17
  physics Physics {
  }
}
Wall {
  translation 8.84657 -12.0461 1.77636e-15
  rotation 0 0 1 1.5708
  name "wall(1)"
  size 0.2 60 2.4
}
PaintBucket {
  translation 1.9000050043557597 -2.7899999015441894 -0.002321064019366291
  rotation 0.016108479407853087 -0.9998702498240288 -2.0199040284741605e-05 0.0038280974816910346
}
Cabinet {
  translation -5.27367 4.11358 0
}
Cabinet {
  translation -5.27367 3.17358 0
  name "cabinet(1)"
}
Blackboard {
  translation -5.4452 -3.3242 1.67
}
Wall {
  translation -5.66824 -5.59486 1.77636e-15
  size 0.2 60 2.4
}
SolidPipe {
  translation -0.15119 2.46511 0.890065
  rotation 0.7161929289331153 0.6979023468003737 5.3635220053471434e-05 0.00030989312636978345
  height 0.3
  radius 0.04
  physics Physics {
  }
}
WoodenBox {
  translation 2.96521 -3.28714 0.3
}
WoodenBox {
  translation 2.35329 -3.28656 0.3
  name "wooden box(1)"
}
WoodenBox {
  translation 2.35329 -3.28656 0.9
  name "wooden box(2)"
}
WoodenBox {
  translation 2.97329 -3.28656 0.9
  name "wooden box(3)"
}
FireExtinguisher {
  translation -5.123450002167625 -5.143850015039682 -0.004708800145291514
  rotation 0.9897802935411524 -0.1426007381383085 -1.039262218323541e-09 0.007893061276697514
}
Table {
  translation 2.05155 1.88815 0
  rotation 0 0 1 1.57
}
Solid {
  translation 2.46 1.51001 0.74079
  rotation 0 0 1 1.5708
  scale 0.015 0.015 0.015
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.1 0.1
        roughness 0.1
        metalness 0.1
      }
      geometry Mesh {
        url [
          "../../../Downloads/creality-ender-3-v2-2.snapshot.5/Ender 3.stl"
        ]
      }
    }
  ]
  boundingObject Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
    mass 1
    centerOfMass [
      0 0 0
    ]
  }
}
Table {
  translation -4.04852 -0.68386 0
  name "table(1)"
}
Table {
  translation -4.06682 1.14437 -1.33227e-15
  name "table(3)"
}
Table {
  translation -0.168007 3.29728 0
  name "table(2)"
  trayAppearance VarnishedPine {
    textureTransform TextureTransform {
      scale 10 10
    }
    IBLStrength 10
  }
}
Table {
  translation 0.240864 1.893 0
  rotation 0 0 1 1.57
  name "table(4)"
  trayAppearance VarnishedPine {
    textureTransform TextureTransform {
      scale 10 10
    }
    IBLStrength 10
  }
}
Door {
  translation -5.6651 -6.60923 2.66454e-15
  position -1.139049985797786e-09
}
IntermodalOfficeContainer {
  translation 0.697281 -10.2307 2.84217e-14
}
RectangleArena {
  translation 2.01068 -5.12401 -3.55271e-14
  floorSize 60 60
  floorAppearance Parquetry {
    type "dark strip"
  }
}
OfficeChair {
  translation -3.129530000485516 1.0801400000172767 -0.00012897002720582695
  rotation 6.327037536348951e-05 -4.4887350054444553e-07 0.9999999979983291 3.1415900000240673
}
OfficeChair {
  translation 0.529845114021313 3.18566986822095 4.94822457324795e-05
  rotation 0.0010538026925732467 1.2328574224572877e-06 0.9999994447490285 3.1415901921098497
  name "office chair(1)"
}
Monitor {
  translation -4.46017 1.05845 0.76
}
Monitor {
  translation -4.46017 1.70845 0.76
  rotation 1 0 0 4.692820414042842e-06
  name "monitor(1)"
}
Robot {
  translation -3.566779568837356 -1.9278490851563985 0.08842778739359006
  rotation 0.008279280608155285 -0.009317268045864637 0.9999223179971397 1.5708723378062173
  children [
    UR5e {
      translation 0.25 0 0.4
      controller "ur5_webot_ctrl"
      toolSlot [
        Robotiq3fGripper {
        }
      ]
      staticBase FALSE
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 0.5
        metalness 0.2
      }
      geometry Mesh {
        url [
          "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/summit_xls_chassis.dae"
        ]
      }
    }
    Solid {
      translation 0 0 0.39062
      children [
        Shape {
          appearance DEF DARKGREY PBRAppearance {
            baseColor 0.5 0.5 0.5
            metalness 0.9
          }
          geometry Mesh {
            url [
              "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/summit_xls_chapas_inox_tapas.stl"
            ]
          }
        }
      ]
      name "base_chapa"
    }
    Solid {
      translation 0.359 0 0.30062
      children [
        Shape {
          appearance USE DARKGREY
          geometry Mesh {
            url [
              "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/summit_xls_chapas_inox_tapas_front.stl"
            ]
          }
        }
      ]
      name "base_chapa_front"
    }
    Solid {
      translation 0 0.28931 0.29062
      children [
        DEF LOGO Shape {
          appearance USE DARKGREY
          geometry Mesh {
            url [
              "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/robotnik_logo_chasis.stl"
            ]
          }
        }
      ]
      name "base_logo_left"
    }
    Solid {
      translation 0 -0.28931 0.29062
      rotation 0 0 1 3.141593
      children [
        USE LOGO
      ]
      name "base_logo_right"
    }
    Solid {
      translation 0.345 0 0
      rotation 0 0 -1 1.570796
      children [
        USE LOGO
      ]
      name "base_logo_front"
    }
    Solid {
      translation -0.345 0 0
      rotation 0 0 1 1.570796
      children [
        USE LOGO
      ]
      name "base_logo_rear"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -213.42455854705796
        axis 0 1 0
        anchor 0.2225 -0.2045 0
      }
      device [
        RotationalMotor {
          name "front_right_wheel_joint"
          acceleration 20
          controlPID 3 0 0
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "front_right_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation 0.22314380396466474 -0.20449996661010172 0.003120120676247796
        rotation 0.9897111242743 0.10117266253847555 -0.10117303416709861 1.5811419875369448
        children [
          DEF INTERIOR_WHEEL Group {
            children [
              Shape {
                appearance DEF WHEEL_APPEARANCE PBRAppearance {
                  baseColor 0.381347 0.381347 0.381347
                  metalness 0.4
                  emissiveColor 0.202 0.202 0.202
                }
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rims_int.dae"
                  ]
                }
              }
              Shape {
                appearance DEF ROLLERS_APPEARANCE PBRAppearance {
                  baseColor 0.08132 0.08132 0.08132
                  roughness 0.8
                  metalness 0
                }
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rollers_int.dae"
                  ]
                }
              }
            ]
          }
        ]
        name "front_right_wheel_link"
        contactMaterial "InteriorWheelMat"
        boundingObject DEF WHEEL_BO Cylinder {
          height 0.08
          radius 0.123
        }
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 935.498581470067
        axis 0 1 0
        anchor 0.2225 0.2045 0
      }
      device [
        RotationalMotor {
          name "front_left_wheel_joint"
          acceleration 20
          controlPID 3 0 0
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "front_left_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation 0.22045778873210523 0.20450003337390876 0.0024429478774319947
        rotation -0.8897353438602345 0.32277720837787244 -0.32277839400693753 4.595818983808608
        children [
          DEF EXTERIOR_WHEEL Group {
            children [
              Shape {
                appearance USE WHEEL_APPEARANCE
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rims_ext.dae"
                  ]
                }
              }
              Shape {
                appearance USE ROLLERS_APPEARANCE
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rollers_ext.dae"
                  ]
                }
              }
            ]
          }
        ]
        name "front_left_wheel_link"
        contactMaterial "ExteriorWheelMat"
        boundingObject USE WHEEL_BO
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -216.4374180380474
        axis 0 1 0
        anchor -0.2225 0.2045 0
      }
      device [
        RotationalMotor {
          name "back_left_wheel_joint"
          acceleration 20
          controlPID 3 0 0
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "back_left_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation -0.22352338932096272 0.2045000329443426 -0.002966319147152917
        rotation 0.11781651214054904 -0.702180762784308 0.7021833420430045 2.9070417447870724
        children [
          USE INTERIOR_WHEEL
        ]
        name "back_left_wheel_link"
        contactMaterial "InteriorWheelMat"
        boundingObject USE WHEEL_BO
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 997.4980322435706
        axis 0 1 0
        anchor -0.2225 -0.2045 0
      }
      device [
        RotationalMotor {
          name "back_right_wheel_joint"
          acceleration 20
          controlPID 3 0 0
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "back_right_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation -0.22563683993500033 -0.20449996703966933 0.00013215791689291258
        rotation -0.5936554946291507 0.5690214297491817 -0.5690235198854396 4.2130721538412175
        children [
          USE EXTERIOR_WHEEL
        ]
        name "back_right_wheel_link"
        contactMaterial "ExteriorWheelMat"
        boundingObject USE WHEEL_BO
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    Solid {
      translation 0.35 0 0.09
      name "base_docking_contact"
    }
    DEF IMU Transform {
      translation 0.127 -0.129 0.212
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.5 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.001 0.001 0.001
          }
        }
        Accelerometer {
        }
        Gyro {
        }
        Compass {
        }
      ]
    }
    DEF RGBD Solid {
      translation 0.3585 0 0.29062
      children [
        Solid {
          translation 0.007 0 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.3 0.3 0.3
                roughness 1
                roughnessMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/plastic/plastic_rough_roughness.jpg"
                  ]
                }
                metalness 0
                normalMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/plastic/plastic_rough_normal.jpg"
                  ]
                }
              }
              geometry Mesh {
                url [
                  "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/orbbec_astra.dae"
                ]
              }
            }
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                roughness 1
                roughnessMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/plastic/plastic_rough_roughness.jpg"
                  ]
                }
                metalness 0
                normalMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/textures/plastic/plastic_rough_normal.jpg"
                  ]
                }
              }
              geometry Mesh {
                url [
                  "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/robotnik/summit_xl_steel/protos/meshes/orbbec_astra_cover.dae"
                ]
              }
            }
          ]
          name "front_rgbd_camera_link"
          boundingObject Transform {
            translation -0.02 0 0.015
            children [
              Box {
                size 0.04 0.165 0.03
              }
            ]
          }
        }
        Camera {
          translation 0 0 0.015
          name "rgb_camera"
          fieldOfView 1.0472
          width 1280
          height 720
        }
        RangeFinder {
          translation 0 0 0.015
          name "depth_camera"
          fieldOfView 1.0472
          width 640
          height 480
          minRange 0.6
          maxRange 8
        }
      ]
    }
    GPS {
      translation -0.365 -0.167 0.302
    }
    Transform {
      translation 0 0 0.39062
    }
    Transform {
      translation 0.2865 -0.20894 0.193
      rotation 0.9238766724085287 -0.3826903371909772 0 3.14159
    }
    Transform {
      translation -0.2865 0.20894 0.193
      rotation 0.3826903371909772 0.9238766724085287 0 3.14159
    }
  ]
  name "Summit-XL Steel"
  boundingObject Group {
    children [
      Transform {
        translation 0 0 0.3
        children [
          Box {
            size 0.719 0.57861 0.18
          }
        ]
      }
      Transform {
        translation 0 0 0.176
        children [
          Box {
            size 0.524 0.27 0.06495
          }
        ]
      }
      Transform {
        translation 0 0 0.046225
        children [
          Box {
            size 0.69 0.27 0.1945
          }
        ]
      }
    ]
  }
  physics Physics {
    density -1
    mass 125
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      1.391 6.853 6.125
      0.004 0 0
    ]
  }
  controller "base_controller"
  controllerArgs [
    ""
  ]
}
